lib/rules/selector-max-compound-selectors/index.cjs

```diff
@@ -67,34 +68,35 @@ const rule = (primary, secondaryOptions) => {
 		 * @param {import('postcss').Rule} ruleNode
 		 */
 		function checkSelector(selectorNode, ruleNode) {
-			let compoundCount = 1;
+			/** @type {import('postcss-selector-parser').Node[]} */
+			const filteredChildNodes = [];
 
-			selectorNode.each((childNode, index) => {
+			selectorNode.each((childNode) => {
 				// Only traverse inside actual selectors and context functional pseudo-classes
-				if (childNode.type === 'selector' || isContextFunctionalPseudoClass(childNode)) {
+				if (isSelector(childNode) || isContextFunctionalPseudoClass(childNode)) {
 					checkSelector(childNode, ruleNode);
 				}
 
-				// Compound selectors are separated by combinators, so increase count when meeting one
-				if (childNode.type === 'combinator') {
-					compoundCount++;
-
-					return;
-				}
-
-				// Try ignoring the selector if the current node is the first, or the previous node is a combinator
-				const previousNode = selectorNode.at(index - 1);
-
-				if ((!previousNode || previousNode.type === 'combinator') && isSelectorIgnored(childNode)) {
-					compoundCount--;
+				if (!isSelectorIgnored(childNode)) {
+					filteredChildNodes.push(childNode);
 				}
 			});
 
-			if (
-				selectorNode.type !== 'root' &&
-				selectorNode.type !== 'pseudo' &&
-				compoundCount > primary
-			) {
+			// Normalize selector nodes and count combinators
+			const combinatorCount = filteredChildNodes
+				.filter((node, i, nodes) => {
+					// Filter out a consecutive combinator
+					return !(isCombinator(node) && i > 0 && isCombinator(nodes[i - 1]));
```

Comment:
question

I might be missing something here but the comment says "consecutive combinator".
AFAIK that be ` `,  `+`, `~` and `>`.
Why would `.foo + + .bar` matter? Isn't that invalid?